packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -344,4 +351,11 @@ function mathExprHasSubmetrics(expr: MetricExpressionConfig) {
   return Object.keys(expr.usingMetrics).length > 0;
 }
 
+function assertSubmetricsCount(expr: MetricExpressionConfig) {
+  if (Object.keys(expr.usingMetrics).length > 10) {
```

Comment:
Thiiiiis... is probably not good enough. Is it really 10 submetrics per math expression? Or is it a limit on the total amount of non-expression metrics that an Alarm is going to retrieve?

Any one of the individual submetrics may itself be a mathexpression that contains yet more submetrics, so you may have a bunch of mathexpressions, each with < 10 submetrics, which render to a list of > 10.

They get rendered to a flat list somewhere at the top. The proper validation is probably to just check the length of that flat list (which I imagine is the same as what the service itself is doing).

----

I see that the documentation says that the current implementation is correct, but I wonder if the docs are simplifying language (to not blow people's heads) past the point of correctness...

Reply:
If you don't feel like experimenting, I guess we leave it like this until someone complains that it doesn't work? :/

Reply:
I think that eventually it is indeed the metrics that the alarm is going to retrieve. Remember though that people can also do `SEARCH(something that retrieves 100 metrics)` - so we don't really have a way of validating that in the CDK.

Since the current validation definitely makes things a little better, preventing a faulty deployment, i'll leave it at that for now, even though we might do better.